File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        vector<int> smap(26, 0), tmap(26, 0);
        for(char c: s) smap[c-'a']++;
        for(char c: t) tmap[c-'a']++;

        /*for(int i = 0; i < 26; i++){
            if(smap[i] != tmap[i]) return false;
        }*/
        if(smap != tmap) return false;

        return true;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        freq = {}
        for i in range(len(s)):
            freq[s[i]] = freq.get(s[i], 0) + 1
        
        for i in range(len(t)):
            if t[i] in freq and freq[t[i]] > 0:
                freq[t[i]] -= 1
            else:
                return False
        
        return True
        